### ISSUE TITLE
Fragments on interfaces cause "Unknown fragment" error

### ISSUE THREAD

**Issue by akosradler** [B2o5T/graphql-eslint#976]
**Describe the bug**

Fragments on interfaces cause "Unknown fragment" error, while using known-fragment-names.

**To Reproduce**

Shema:

```gql
interface Bird {
  id: String!
  name: String!
}

type Penguin implements Bird {
  id: String!
  name: String!
}

type Picture_of_a_penguin {
  id: String!
  penguin: Penguin!
  imgUrl: String!
}
```

document:

```gql
fragment BirdFragment on Bird {
  id: String!
  name: String!
}

query getPenguinPicture {
  picture_of_a_penguin() {
    id
    penguin {
      ...BirdFragment
    }
  }
}

```

**Expected behavior**

I would expect this not to be an error, since the fragment is a subset of what is being queried.

**Environment:**

- OS: Windows
- `@graphql-eslint: 3.8.0`:
- NodeJS: 16.13.0

**Comment by B2o5T** [comment]
Hi @akosradler, sorry for the long response, I couldn't reproduce your error 🤷‍♂️.
Can you create a reproduction repo with minimal dependencies?

Operation
```gql
fragment BirdFragment on Bird {
  id
  name
}

query getPenguinPicture {
  picture_of_a_penguin {
    id
    penguin {
      ...BirdFragment
    }
  }
}
```

Schema
```gql
interface Bird {
  id: String!
  name: String!
}

type Penguin implements Bird {
  id: String!
  name: String!
}

type Picture_of_a_penguin {
  id: String!
  penguin: Penguin!
  imgUrl: String!
}

type Query {
  picture_of_a_penguin: Picture_of_a_penguin
}
```

**Comment by akosradler** [comment]
Hi @B2o5T 
No worries, here's a reproduction repo.
https://github.com/akosradler/reslint-graphql-unknown-fragment

Everything is in the readme, but feel free to reach out.

Thank you

**Comment by B2o5T** [comment]
Thanks for the reproduction, it's a similar issue like in <https://github.com/B2o5T/graphql-eslint/issues/983>
I will fix it in the next version of `graphql-eslint` with covered tests 👍

**Comment by B2o5T** [comment]
fixed in [@graphql-eslint/eslint-plugin@3.9.1](https://github.com/B2o5T/graphql-eslint/releases/tag/%40graphql-eslint%2Feslint-plugin%403.9.1)
